### ISSUE TITLE
Mobile App not getting published properly

### ISSUE THREAD

**Issue by krunalc** [Azure/azure-mobile-apps-net-server#105]
I have implemented Azure Mobile App and Xamarin.Forms Client application. I want user to login using facebook from Phone and also want to fetch user's profile data. For this I have implemented the additional call/method into API controller in Azure Mobile App. I have followed steps and put the code as per your article but somehow get following error message when I run the Mobile App on localhost or trying to publish 

**Multiple types were found that match the controller named 'Home'. This can happen if the route that services this request ('') found multiple controllers defined with the same name but differing namespaces, which is not supported. The request for 'Home' has found the following matching controllers: Microsoft.Azure.Mobile.Server.Controllers.HomeController Microsoft.WindowsAzure.Mobile.Service.Controllers.HomeController
**

I understand this is related config settings. I have following code in place 

`            HttpConfiguration config = new HttpConfiguration();

            new MobileAppConfiguration()
                        .UseDefaultConfiguration()
                        .ApplyTo(config);

            app.UseWebApi(config);
            app.UseAppServiceAuthentication(new AppServiceAuthenticationOptions
            {
                SigningKey = ConfigurationManager.AppSettings["SigningKey"],
                ValidAudiences = new[] { ConfigurationManager.AppSettings["ValidAudience"] },
                ValidIssuers = new[] { ConfigurationManager.AppSettings["ValidIssuer"] },
                TokenHandler = config.GetAppServiceTokenHandler()
            });
`

If I remove the default configuration from above then exception message go away but in that case I don't see the app getting hosted properly i.e. it is showing blank page in browser instead of ready page shown once app is hosted properly. 

I'm getting no help on this and searching on this since last two days without any success. 

I hope someone can provide some help/pointers to fix this issue.

Regards,
Krunal

**Comment by brettsam** [comment]
`Microsoft.WindowsAzure.Mobile.Service.Controllers.HomeController` comes from the Mobile Services SDK, but you're trying to use Mobile Apps. Can you remove any nuget references for `WindowsAzure.MobileServices.Backend.*`?

**Comment by krunalc** [comment]
Hi brettsam,

Yup you are correct in pointing out the problem. If I don't have WindowAzure.MobileServices.Backend.? then this problem won't occur. But if I don't add these references, I won't get "ServiceUser" and without "ServiceUser" I can't find any way to get the profile data for Facebook/Google etc.

Actually what I want to achieve is to implement the FB/Google login and fetch the profile data. I get logged on User's token (i.e. I guess it's Azure AD token) but I can't get the profile data for FB and google using that token. So I have to cast the user to ServiceUser and call the "GetIdentitiesAsyn" as various posts suggest. http://stackoverflow.com/questions/32390377/get-user-info-like-name-email-id-etc-from-authentication-token-in-net-backend

So would appreciate if any help/pointer is provided with above requirement in mind.

Thanks a lot,
Krunal

**Comment by brettsam** [comment]
For Mobile Apps, `ServiceUser` is no longer needed. Instead you can follow the directions here: https://github.com/Azure/azure-mobile-apps-net-server/wiki/Understanding-User-Ids

In short, you can now do this: `creds = await this.User.GetAppServiceIdentityAsync<FacebookCredentials>(this.Request);`. I believe you may have to add `using System.Security.Principal;` to your class to get the `GetAppServiceIdentityAsync` extension method to appear.

**Comment by krunalc** [comment]
Thanks brett for link and some explanation. I will work on it and let you know the outcome

**Comment by adrianhall** [comment]
Seems like this is resolved - closing.

**Comment by krunalc** [comment]
Hello Adrian/brett,

I had to pause working on this thing in between and now I'm starting on it again. I checked the [link ](https://github.com/Azure/azure-mobile-apps-net-server/wiki/Understanding-User-Ids) you have shared. In the code snippet , in the very first line "this.User" is cast to "ClaimsPrincipal". I wonder what is this "**this.User**" on controller side?

On the client side, after successful login, I get "MobileServiceUser". Do I need to send this user to API call (i.e. controller side) and cast it to claimsprinciple to proceed further?

Pls clarify,

Thanks,
Krunal
